### PR TITLE
skip tests Gd callback to nih

### PR DESCRIPTION
**Pull request by gdlex4015**
Added callback to NIH function and using it to add users to auth domain each time the script is ran

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Review by aweng98 — APPROVED** [review]

File: automation/submission-perf-test.sh [hunk 1]
```diff
@@ -9,13 +9,33 @@ WORKING_DIR=${3:-$PWD}
...
+    user=$1
+
+    echo "
+    Launching calback to NIH for:
+    user=$1
+    "
+     ACCESS_TOKEN=`docker run --rm -v $WORKING_DIR:/app/populate -w /app/populate broadinstitute/dsp-toolbox python get_bearer_token.py "${user}" "${JSON_CREDS}"`
+
+    # Verify that user does not need to refresh their token
+        if
```
Comment (andy7i) [inline automation/submission-perf-test.sh]: This block of code for refresh tokens is duplicated in this script. Delete one of them. It makes sense to check all access issues before running the script so delete the other block. It's also not related to the NIH call so either it shouldn't be in this function or it should be renamed.

File: automation/submission-perf-test.sh [hunk 2]
```diff
@@ -40,27 +60,32 @@ launchSubmission() {
         expression=$9
     "
 
-    ACCESS_TOKEN=`docker run --rm -v $WORKING_DIR:/app/populate -w /app/populate broadinstitute/dsp-toolbox python get_bearer_token.py "${user}" "${JSON_CREDS}"`
+     ACCESS_TOKEN=`docker run --rm -v $WORKING_DIR:/app/populate -w /app/populate broadinstitute/dsp-toolbox python get_bearer_token.py "${user}" "${JSON_CREDS}"`
 
     # Verify that user does not need to refresh their token
-    if curl -f -v --silent -X GET --header "Accept: application/json" --header "Authorization: Bearer $ACCESS_TOKEN" "https://firecloud-orchestration.dsde-alpha.broadinstitute.org/api/refresh-token-status"  2>&1 | grep '"requiresRefresh": true'
-    then
+        if
```
Comment (andy7i) [inline automation/submission-perf-test.sh]: nitpick: indentation

File: automation/submission-perf-test.sh [hunk 3]
```diff
@@ -40,27 +60,32 @@ launchSubmission() {
...
     if [ -z ${9+x} ] ; then
         curl -f "https://firecloud-orchestration.dsde-$ENV.broadinstitute.org/api/workspaces/$namespace/$name/submissions" -H "origin: https://firecloud.dsde-$ENV.broadinstitute.org" -H "accept-encoding: gzip, deflate, br" -H "authorization: Bearer $ACCESS_TOKEN" -H "content-type: application/json" --data-binary "{\"methodConfigurationNamespace\":\"$methodConfigurationNamespace\",\"methodConfigurationName\":\"$methodConfigurationName\",\"entityType\":\"$entityType\",\"entityName\":\"$entityName\",\"useCallCache\":$useCallCache}" --compressed
     else
         curl -f "https://firecloud-orchestration.dsde-$ENV.broadinstitute.org/api/workspaces/$namespace/$name/submissions" -H "origin: https://firecloud.dsde-$ENV.broadinstitute.org" -H "accept-encoding: gzip, deflate, br" -H "authorization: Bearer $ACCESS_TOKEN" -H "content-type: application/json" --data-binary "{\"methodConfigurationNamespace\":\"$methodConfigurationNamespace\",\"methodConfigurationName\":\"$methodConfigurationName\",\"entityType\":\"$entityType\",\"entityName\":\"$entityName\",\"useCallCache\":$useCallCache,\"expression\":\"$expression\"}" --compressed
     fi
-}
+    }
```
Comment (andy7i) [inline automation/submission-perf-test.sh]: nitpick: indentation

File: automation/submission-perf-test.sh [hunk 4]
```diff
@@ -40,27 +60,32 @@ launchSubmission() {
...
 
 
 if [ $ENV = "alpha" ]; then
+    callbackToNIH harry.potter@test.firecloud.org
```
Comment (andy7i) [inline automation/submission-perf-test.sh]: Instead of calling the checking function multiple times for each user, it would be better to call it once for all users. The function can have a loop that checks for each parameter (user). Then after all users have been checked, it should `exit 1` if any of the users did not have their refresh token.

**Review by andy7i — CHANGES_REQUESTED** [review]
I like the idea of checking everything at the beginning. Some improvements will check all users before exiting the script.

File: automation/submission-perf-test.sh (hunk 1 above)
Comment (gdlex4015) [reply automation/submission-perf-test.sh]: Don't we need to get access token for the user?

Comment (andy7i) [reply automation/submission-perf-test.sh]: (we face talked -- this is for the refresh token)

**Comment by gdlex4015** [comment]
skip tests

File: automation/submission-perf-test.sh [hunk 5]
```diff
@@ -9,13 +9,50 @@ WORKING_DIR=${3:-$PWD}
 
 # Check if $ENV is not empty
 if [[ ! ${ENV} ]]; then
-    echo "ENV is empty. Run again \`sh submission-perf-test.sh <alpha or staging>\`"
+    echo
+    "ENV is empty. Run again sh submission-perf-test.sh <alpha or staging"
```
Comment (andy7i) [inline automation/submission-perf-test.sh]: why was this line changed?

Comment (andy7i) [reply automation/submission-perf-test.sh]:
``` 
-    echo "ENV is empty. Run again \`sh submission-perf-test.sh <alpha or staging>\`"
+    echo
+    "ENV is empty. Run again sh submission-perf-test.sh <alpha or staging"
```

File: automation/submission-perf-test.sh [hunk 6]
```diff
@@ -9,13 +9,50 @@ WORKING_DIR=${3:-$PWD}
...
+callbackToNIH() {
+    user=$1
+
+    echo "
+    Launching calback to NIH for:
+    user=$1
+    "
+     ACCESS_TOKEN=`docker run --rm -v $WORKING_DIR:/app/populate -w /app/populate broadinstitute/dsp-toolbox python get_bearer_token.py "${user}" "${JSON_CREDS}"`
+
+      curl -X POST --header "Content-Type: application/json" --header "Accept: application/json" --header "Authorization: Bearer $ACCESS_TOKEN" -d "{\"jwt\":\"eyJhbGciOiJIUzI1NiJ9.ZmlyZWNsb3VkLWRldg.NPXbSpTmAOUvJ1HX85TauAARnlMKfqBsPjumCC7zE7s\"}" "https://firecloud-orchestration.dsde-$ENV.broadinstitute.org/api/nih/callback"
```
Comment (andy7i) [inline automation/submission-perf-test.sh:54]: Might want to check with security folks that it's okay that we're putting a jwt into a public(?) github repo.

File: automation/submission-perf-test.sh [hunk 7]
```diff
@@ -40,27 +77,25 @@ launchSubmission() {
...
 
-
 if [ $ENV = "alpha" ]; then
+for user in "${users[@]}"
+    do
+        checkToken $user
+    done
+    for user in "${users[@]}"
+    do
+        callbackToNIH $user
```
Comment (andy7i) [inline automation/submission-perf-test.sh]: If we keep this implementation the way it is, you could put both checkToken and callbackToNIH into one for loop

File: automation/submission-perf-test.sh [hunk 8]
```diff
@@ -9,13 +9,50 @@ WORKING_DIR=${3:-$PWD}
...
+checkToken () {
+    user=$1
+
+    # Verify that user does not need to refresh their token
+    if
+        curl -f -v --silent -X GET --header "Accept: application/json" --header "Authorization: Bearer $ACCESS_TOKEN" "https://firecloud-orchestration.dsde-$ENV.broadinstitute.org/api/refresh-token-status"  2>&1 | grep '"requiresRefresh": true'
+    then
+        echo "This user needs its refresh token refreshed"
+        exit 1
+    fi
```
Comment (andy7i) [inline automation/submission-perf-test.sh]: As mentioned previously, I think it would be useful for the code to check ALL users' tokens in `checkToken`. If any need refreshing, then print out the names of each user and then `exit 1`.

**Review by andy7i — COMMENTED** [review]
As mentioned previously, I think it would be useful for the code to check ALL users' tokens in `checkToken`. If any need refreshing, then print out the names of each user and then `exit 1`.

File: automation/submission-perf-test.sh (hunk 6 above)
Comment (gdlex4015) [reply automation/submission-perf-test.sh:54]: It is, we had this discussion while we did the same with swatomation, this JWT is only for our internal domains

File: automation/submission-perf-test.sh (hunk 7 above)
Comment (gdlex4015) [reply automation/submission-perf-test.sh]: I don't want to do so, if user needs token refresh I want the script to stop without going further

File: automation/submission-perf-test.sh (hunk 8 above)
Comment (gdlex4015) [reply automation/submission-perf-test.sh]: I agree with that, will change

File: automation/submission-perf-test.sh (hunk 7 above)
Comment (andy7i) [reply automation/submission-perf-test.sh]: I think it would eliminate duplicate code. I think we can do both in one for loop.

Also, we can stick this check before ` if [ $ENV = "alpha" ]; then`. (Both alpha and staging use the same users so it just needs to be run once before either of them run. )

Also, indentation ;)

**Review by andy7i — APPROVED** [review]
A suggestion for eliminating duplicate code, then 👍

**Comment by gdlex4015** [comment]
I am going to merge, but let's pick it up tomorrow on the duplicate code thing.
